### PR TITLE
[tckrefactor] Jakarta Persistence 3.2 - JPQL new feature plus Persistence TCK runner and stored proc fixes

### PR DESCRIPTION
**Pull request by rfelcman**
Another set of tests for new Jakarta Persistence 3.2. In this case it's about JPQL new features:
https://jakartaee.github.io/persistence/latest-nightly/nightly.html#jakarta-persistence-3-2

- Added union, intersect, except, cast, left, right, and replace for Jakarta Persistence QL and criteria queries
- Added || string concatenation operator to Jakarta Persistence QL
- Added support for specifying null precedence when ordering Jakarta Persistence QL and criteria queries
https://jakartaee.github.io/persistence/latest-nightly/nightly.html#a5587

There are two new improvements/fixes:
- In `*persistence.sprocs.sql` Java package changes to `CS_Procs` class to catch up last refactoring in the project
- In `jpa/bin/pom.xml` to make it runnable like add missing dependencies, path fix and some refactor like version properties and put eclipselink part into Maven profile

CC @lukasj @alwin-joseph @anajosep @arjantijms @cesarhernandezgt @dblevins @m0mus @edbratt @gurunrao @jansupol @jgallimore @kazumura @kwsutter @LanceAndersen @bhatpmk @RohitKumarJain @shighbar @gthoman @brideck @OndroMih @dmatej
@starksm64 @scottmarlow

### CODE REVIEW

File: jpa/bin/pom.xml [hunk 1]
```diff
@@ -15,20 +15,30 @@
...
+        <jpa.classes>
+            ${glassfish.module.dir}/jakarta.persistence-api.jar,${glassfish.module.dir}/org.eclipse.persistence.jpa.jar,${glassfish.module.dir}/jakarta.servlet-api.jar,${glassfish.module.dir}/jakarta.transaction-api.jar,${glassfish.module.dir}/jakarta.ejb-api.jar,${glassfish.module.dir}/org.eclipse.persistence.antlr.jar,${glassfish.module.dir}/org.eclipse.persistence.asm.jar,${glassfish.module.dir}/org.eclipse.persistence.core.jar,${glassfish.module.dir}/org.eclipse.persistence.dbws.jar,${glassfish.module.dir}/org.eclipse.persistence.jpa.jpql.jar,${glassfish.module.dir}/org.eclipse.persistence.jpa.modelgen.processor.jar,${glassfish.module.dir}/org.eclipse.persistence.moxy.jar,${glassfish.module.dir}/org.eclipse.persistence.oracle.jar,${glassfish.module.dir}/jakarta.validation-api.jar,${glassfish.module.dir}/hibernate-validator-cdi.jar,${glassfish.module.dir}/hibernate-validator.jar
+        </jpa.classes>
```
Comment (lukasj) [inline jpa/bin/pom.xml:40]: this particular property may need to be defined twice - once for container testing, second time for SE testing;

There is a test for SE which expects no implementation of Validation API. It's the one currently failing and @gurunrao is working on that

File: jpa/bin/pom.xml [hunk 2]
```diff
@@ -265,8 +290,11 @@
                         </goals>
                         <configuration>
                             <includes>
+<!--
                                 <include>**/*Client*.*</include>
                                 <include>**/*JPASigTest*.*</include>
+-->
+                                <include>ee.jakarta.tck.persistence.core.query.language.Client3</include>
```
Comment (lukasj) [inline jpa/bin/pom.xml:297]: perhaps uncomment the section commented out  and remove this new line?

File: jpa/bin/pom.xml [hunk 3]
```diff
@@ -324,4 +373,142 @@
...
+                    <artifactId>org.eclipse.persistence.asm</artifactId>
+                    <version>${eclipselink.asm.version}</version>
+                </dependency>
+            </dependencies>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-dependency-plugin</artifactId>
+                        <version>3.2.0</version>
```
[the remark this answers is not in the document]
Comment (rfelcman) [reply jpa/bin/pom.xml]: `pluginManagement` section created + plugins updated to the latest available version
I agree with comment about another improvements in the pom. I want just something functional on the beginning.
This is something like 0.5.0 version.

**Review by lukasj — APPROVED** [review]
LGTM; note there might be some collisions with #1249

**Pull request merged**